code/game/machinery/recharger.dm

```diff
@@ -163,3 +167,11 @@
 			icon_state = "recharger2"
 		return
 	icon_state = "recharger0"
+
+obj/machinery/recharger/wallcharger
+	name = "wall recharger"
+	icon_state = "wrecharger0"
+	icon_state_off = "wrechargeroff"
+	icon_state_idle = "wrecharger0"
+	icon_state_charging = "wrecharger1"
+	icon_state_charged = "wrecharger2"

```

Comment:
Did you even test this in game? This isn't how icon_states work.

Also if rechargers are going to have subtypes, split their icons into a new file altogether.